### ISSUE TITLE
BatchNorm2d でのバッチサイズ1

### ISSUE THREAD

**Issue by hkawash** [YutaroOgawa/pytorch_advanced#186]
小川様

バッチノーマリゼーションについての質問です．

nn.BatchNorm2d を使ったネットワークで学習するいくつかのコードで，ミニバッチがサイズ1だとエラーになるのでさける，となっています．BatchNorm2d は Spatial Batch Normalization なので，統計量は N x H x W  のデータから計算されるため，N=1 でもエラーは起きないようにも思いました（実際手元の環境では4章のコードはエラーが生じないようでした）．
バージョンにもよるでしょうか．念のためご確認いただければ幸いです．

**Comment by hkawash** [comment]
ご確認いただきどうもありがとうございました．drop_last の情報についてもどうもありがとうございます．
